wps_tools/utils.py

```diff
@@ -12,9 +12,80 @@
 import logging
 import os
 from pathlib import Path
+from urllib.parse import urlparse
+from urllib.request import urlretrieve
+from collections import OrderedDict
 
 MAX_OCCURS = 1000
 
+common_status_percentages = {
+    "start": 0,
+    "process": 20,
+    "build_output": 95,
+    "complete": 100,
+}
+
+
+def url_handler(workdir, url):
+    """Handles URL based on its type
+
+    A process cannot access to the data from an HTTPServer URL without downloading
+    while OPeNDAP URL can be treated as a filepath.
+    The function returns the given URL if it is an OPeNDAP path.
+    Otherwise, data from the HTTPServer URL are copied to a file created in workdir,
+    and the path to the file is returned.
+
+    Parameters:
+        workdir (str): Path to the workdir
+        url (str): URL to be handled
+
+    Returns:
+        url/local_file (str): URL/filepath with accessible data
+    """
+    if is_opendap_url(url):
+        # OPeNDAP
+        return url
+    elif urlparse(url).scheme and urlparse(url).netloc:
+        # HTTPServer or other
+        local_file = os.path.join(workdir, url.split("/")[-1])
+        urlretrieve(url, local_file)
+        return local_file
+
+
+def collect_args(request, workdir):
```

Comment:
It may be challenging but this may have to be even more robust. We can have cases where there are multiple items in a single input. Right now this method **only** looks at the `0` index.